pkg/agent/plugin/workloadattestor/k8s/sigstore/sigstorecache_test.go

```diff
@@ -0,0 +1,444 @@
+		Key: "signature2",
+		Value: []SelectorsFromSignatures{
+			{
+				Subject:        "spirex2@example.com",
+				Content:        "content5",
+				LogID:          "5555555555555555",
+				IntegratedTime: "5555555555555555",
+			},
+		},
+	}
+)
+
+func TestNewCache(t *testing.T) {
+	want := &cacheImpl{
+		size:     3,
+		items:    list.New(),
+		mutex:    sync.RWMutex{},
+		itemsMap: make(map[string]MapItem),
+	}
+	got := NewCache(3)
+	require.Equal(t, want, got, "NewCache() = %v, want %v", got, want)
+}
+
+func TestCacheimpl_GetSignature(t *testing.T) {
+	m := make(map[string]MapItem)
+	items := list.New()
+
+	m[selectors1.Key] = MapItem{
+		item:    &selectors1,
+		element: items.PushFront(selectors1.Key),
+	}
+	m[selectors2.Key] = MapItem{
+		item:    &selectors2,
+		element: items.PushFront(selectors2.Key),
+	}
+
+	cacheInstance := &cacheImpl{
+		size:     3,
+		items:    items,
+		mutex:    sync.RWMutex{},
+		itemsMap: m,
+	}
+
+	tests := []struct {
+		name         string
+		want         *Item
+		key          string
+		errorMessage string
+	}{
+		{
+			name:         "Non existing entry",
+			want:         nil,
+			key:          selectors3.Key,
+			errorMessage: "A non-existing item's key should return a nil item.",
+		},
+		{
+			name:         "Existing entry",
+			want:         &selectors1,
+			key:          selectors1.Key,
+			errorMessage: "An existing items key's should return the existing item",
+		},
+	}
+
+	for _, tt := range tests {
+		got := cacheInstance.GetSignature(tt.key)
+		require.Equal(t, got, tt.want, "%v Got: %v Want: %v", tt.errorMessage, got, tt.want)
+	}
+}
+
+func TestCacheimpl_PutSignature(t *testing.T) {
+	m := make(map[string]MapItem)
+	items := list.New()
+
+	cacheInstance := &cacheImpl{
+		size:     2,
+		items:    items,
+		mutex:    sync.RWMutex{},
+		itemsMap: m,
+	}
+
+	tests := []struct {
+		name       string
+		item       *Item
+		wantLength int
+		wantKey    string
+		wantValue  *Item
+	}{
+		{
+			name:       "Put first element",
+			item:       &selectors1,
+			wantLength: 1,
+			wantKey:    selectors1.Key,
+			wantValue:  &selectors1,
+		},
+		{
+			name:       "Put first element again",
+			item:       &selectors1,
+			wantLength: 1,
+			wantKey:    selectors1.Key,
+			wantValue:  &selectors1,
+		},
+		{
+			name:       "Put second element",
+			item:       &selectors2,
+			wantLength: 2,
+			wantKey:    selectors2.Key,
+			wantValue:  &selectors2,
+		},
+		{
+			name:       "Overflow cache",
+			item:       &selectors3,
+			wantLength: 2,
+			wantKey:    selectors3.Key,
+			wantValue:  &selectors3,
+		},
+		{
+			name:       "Update entry",
+			item:       &selectors3Updated,
+			wantLength: 2,
+			wantKey:    selectors3.Key,
+			wantValue:  &selectors3Updated,
+		},
+	}
+
+	for _, tt := range tests {
+		cacheInstance.PutSignature(*tt.item)
+		gotLen := cacheInstance.items.Len()
+		if gotLen != tt.wantLength {
+			t.Errorf("Item count should be %v in test case %q", tt.wantLength, tt.name)
+		}
```

Comment:
```suggestion
                 require.Len(t, cacheInstance.items, tt.WantLength)
```

Reply:
`%!s(int=1)}" could not be applied builtin len()`
[link for failed CI](https://github.com/willallves/spire/actions/runs/3221520111/jobs/5269567154#step:5:76)
Reversed commit